crates/db_schema/src/impls/post.rs

```diff
@@ -240,17 +225,33 @@ impl Post {
     let pictrs_search = "%pictrs/image%";
 
     diesel::update(
-      post
-        .filter(community_id.eq(for_community_id))
-        .filter(url.like(pictrs_search)),
+      post::table
+        .filter(post::community_id.eq(for_community_id))
+        .filter(post::url.like(pictrs_search)),
     )
     .set((
-      url.eq::<Option<String>>(None),
-      thumbnail_url.eq::<Option<String>>(None),
+      post::url.eq::<Option<String>>(None),
+      post::thumbnail_url.eq::<Option<String>>(None),
     ))
     .get_results::<Self>(conn)
     .await
   }
+
+  /// Lists local community ids for all posts for a given creator.
+  pub async fn list_creators_local_community_ids(
+    pool: &mut DbPool<'_>,
+    for_creator_id: PersonId,
+  ) -> Result<Vec<CommunityId>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    post::table
+      .inner_join(community::table)
+      .filter(community::local.eq(true))
+      .filter(post::creator_id.eq(for_creator_id))
+      .select(community::id)
+      .distinct()
+      .load::<CommunityId>(conn)
+      .await
```

Comment:
You could move this into person and use a single function for both post and comment, maybe even a single sql query. Doesnt make much sense to have two separate functions.

The function name is also not so clear, how about `list_user_posted_in_communities()`?

Reply:
K I moved these to Person, and used a `union` query.